RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -143,6 +263,45 @@ extension PurchaseHandler {
         }
     }
 
+    @MainActor
+    func performExternalRestoreLogic() async throws -> (info: CustomerInfo, success: Bool) {
+        Logger.debug(Strings.executing_external_restore_logic)
+
+        self.restoreInProgress = true
+        self.restoredCustomerInfo = nil
+        self.restoreError = nil
+
+        DispatchQueue.main.async {
+            // this triggers the view's `.handleRestore` function, and its callback must be called
+            // after the continuation is set below
+            self.performRestore = RestoreResultReporter(callback: self.reportExternalRestoreResult)
+        }
+
+        self.startAction()
+
+        defer {
+            self.restoreInProgress = false
+            self.actionInProgress = false
+        }
```

Comment:
Nit but can we move defer up to the top of the function so that it gets set if any issues happens in any of the other things about it currently?